src/dash-table/components/Table/index.tsx

```diff
@@ -49,24 +50,45 @@ export default class Table extends Component<SanitizedAndDerivedProps, Standalon
     }
 
     componentWillReceiveProps(nextProps: SanitizedAndDerivedProps) {
-        if (nextProps.filter_query === this.props.filter_query) {
-            return;
-        }
-
         this.setState(state => {
-            const { workFilter: { map: currentMap, value } } = state;
-
-            if (value !== nextProps.filter_query) {
-                const map = this.filterMap(
-                    currentMap,
-                    nextProps.filter_query,
-                    nextProps.visibleColumns
-                );
-
-                return map !== currentMap ? { workFilter: { map, value} } : null;
-            } else {
-                return null;
+            const { applyFocus: currentApplyFocus, workFilter: { map: currentMap, value } } = state;
+
+            const nextState: Partial<StandaloneState> = {};
+
+            // state for filter
+            if (nextProps.filter_query !== this.props.filter_query) {
+                if (value !== nextProps.filter_query) {
+                    const map = this.filterMap(
+                        currentMap,
+                        nextProps.filter_query,
+                        nextProps.visibleColumns
+                    );
+
+                    if (map !== currentMap) {
+                        nextState.workFilter = { map, value };
+                    }
+                }
+            }
+
+            // state for applying focus
+            if (nextProps.active_cell !== this.props.active_cell) {
+                if (!currentApplyFocus) {
+                    nextState.applyFocus = true;
+                }
+            } else if (nextProps.loading_state !== this.props.loading_state) {
```

Comment:
you're going to have to explain this one to me... why does entering *or* exiting loading trigger this check?

Reply:
If a callback modifying `data` is triggered and the focus is on a table cell, the table needs to be re-rendered once the callback returns to make the cell writeable again.